app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/TransactionAnalysisDelegate.kt

```diff
@@ -134,47 +142,59 @@ class TransactionAnalysisDelegate @Inject constructor(
         }
     }
 
-    private suspend fun processConformingManifest(transactionType: TransactionType): PreviewType {
+    private suspend fun ExecutionSummary.processConformingManifest(): PreviewType {
         val networkId = requireNotNull(getProfileUseCase.currentNetwork()?.knownNetworkId)
         val xrdAddress = XrdResource.address(networkId)
-        val resources = getResourcesUseCase(addresses = transactionType.involvedResourceAddresses + xrdAddress).getOrThrow()
+        val transactionType = detailedClassification.first()
+        val resources = getResourcesUseCase(addresses = involvedResourceAddresses + xrdAddress).getOrThrow()
 
         return when (transactionType) {
-            is TransactionType.GeneralTransaction -> transactionType.resolve(
+            is DetailedManifestClass.General -> transactionType.resolve(
                 resources = resources,
                 getTransactionBadgesUseCase = getTransactionBadgesUseCase,
                 getProfileUseCase = getProfileUseCase,
-                resolveDAppsUseCase = resolveDAppsUseCase
+                resolveDAppsUseCase = resolveDAppsUseCase,
+                executionSummary = this
             )
 
-            is TransactionType.SimpleTransfer -> transactionType.resolve(
-                getProfileUseCase = getProfileUseCase,
-                resources = resources
-            )
-
-            is TransactionType.Transfer -> transactionType.resolve(
-                getProfileUseCase = getProfileUseCase,
-                resources = resources
-            )
-
-            is TransactionType.AccountDepositSettings -> transactionType.resolve(
+            is DetailedManifestClass.AccountDepositSettingsUpdate -> transactionType.resolve(
                 getProfileUseCase = getProfileUseCase,
                 allResources = resources
             )
 
-            is TransactionType.StakeTransaction -> {
+            is DetailedManifestClass.Transfer -> resolveTransfer(getProfileUseCase, resources)
```

Comment:
So far in this PR the extension methods where using a receiver to be the transaction type. Although this method does not. Why is it different? If it is unused, maybe the receiver in all methods should change to be the execution summary and not the transaction type, and pass the transaction type as a parameter

Reply:
Because for transfer transaction type does not carry any relevant information, all needed to resolve is in ExecutionSummary. On the other hand it is convenient to do the check for transaction type here, since different types need different data to be pre-fetched. If receiver would be Execution summary and type check would happen inside, I will also need to pass in all the use cases there and do data fetching in that extension function